### PR TITLE
Add the Fun module

### PR DESCRIPTION
**Pull request by Dean177**
This PR Adds a `Fun` module providing some useful functions (and providing a home for a couple of existing ones).

This is initially just for the native side, but if the API looks good I will add tests and the bs side of things.

### CODE REVIEW

File: native/src/tablecloth.mli [hunk 1]
```diff
@@ -1,10 +1,147 @@
...
+  val composeRight : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
+  (** Function composition, passing results along in the suggested direction. For
+example, the following code checks if the square root of a number is odd:
+
+    {[Int.squareRoot >> Int.isEven >> not]} *)
+
+  val ( >> ) : ('a -> 'b) -> ('b -> 'c) -> 'a -> 'c
+  (** See {!Fun.composeRight} *)
+
+  val tap : 'a -> f:('a -> unit) -> 'a
```
Comment (pbiggar) [inline native/src/tablecloth.mli]: What does tap mean?

Comment (Dean177) [reply native/src/tablecloth.mli]: It means "tap into"

Its based on:
- https://apidock.com/ruby/Object/tap
- https://lodash.com/docs/4.17.15#tap

**Comment by Dean177** [comment]
@pbiggar this is now complete for bs & native if you are happy with the naming / docs / API

**Comment by pbiggar** [comment]
Nice work, thank you!

**Pull request merged**